lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -943,7 +943,7 @@ def get_final_rules(client, module, security_group_rules, specified_rules, purge
                         format_rule['user_id_group_pairs'][0].pop(k)
             final_rules.append(format_rule)
             # Order final rules consistently
-            final_rules.sort(key=lambda x: x.get('cidr_ip', x.get('ip_ranges', x.get('ipv6_ranges', x.get('prefix_list_ids', x.get('user_id_group_pairs'))))))
+            final_rules.sort(key=lambda x: json.dumps(x, sort_keys=True))
```

Comment:
I think you can probably do this instead:
```suggestion
            final_rules = json.loads(json.dumps(final_rules, sort_keys=True))
```